pkg/backend/backend.go

```diff
@@ -351,6 +351,16 @@ func (s *BackendServer) GetMatchedRule(rules []models.ContentRule, req *models.R
 		return matchedRules[0], nil
 	}
 
+	// Check if honeypot has a preferred content ID that matches one of the rules
```

Comment:
Can you undo this for now in this PR and move it to a new one? I prefer we don't add more functionality to the PR.  Ideally for things like this we create an issue, describe the problem (and possible solution) and then create a PR specifically and only for that issue.  That keeps things clean and easier to review.